docs/articles/immutability-in-js.md

```diff
@@ -0,0 +1,108 @@
+>
+> -- <cite>[Alexander Kondov](https://hackernoon.com/functional-programming-paradigms-in-modern-javascript-immutability-4e9751ca005c)</cite>
+
+It has a declarative approach of programming, which means that you focus on describing what your program must accomplish rather than how it should do it. It gives more meaning to your code so the next developper will understand it more easily. Functional programming brings along other concepts that help reach this goal, such as immutability.
+
+### What are the benefits?
+
+Does it sound like a hype term to you? Immutabilty brings many solutions to programming matters we ecounter everyday:
+
+- Avoid side effects
+- Data changes detection made simple (shallow comparison)
+- Explicit data changes
+- Memoization
+- Memory optimization
+- Better rendering performances
+- Easy testing
+
+> « Unlike most trends in the world of JavaScript, data immutability is bound to stick with us for a while, and for good reason: firstly, because it’s not a trend: it’s a way of coding (and thinking in code) that promotes clarity, ease of use and understanding data flow, and makes code less prone to errors. »
+>
+> -- <cite>[Ricardo Magalhães](https://hackernoon.com/data-immutability-with-vanilla-javascript-63834a65a6c9)</cite>
+
+In the last few years one of our biggest challenges has been to find an efficient way to detect changes in our data to determine whether or not we should render our interfaces. It's easy to detect changes between primitive values, but it's a completely different issue for objects and arrays. You could compare them by value but you would have to implement recursive algorithms and deal with potential issues like cyclical references. Another method would be to compare object references with the strict equality operator `===`. It's more effective and there isn't any risk to enter in some deathly infinity loop. That's why modern frameworks enforce this concept.
+
+### Highlighted by modern frameworks/libraries
+
+Modern frontend frameworks and libraries are based on a concept that improves drastically performances. This is the well-known Virtual DOM. This technology has been created from a simple evidence: DOM manipulations are expensive.
+
+For the benefits explained in the previous sections, frontend frameworks and libraries chose to use immutability in order to improve their performances. Nowadays we have to deal with more and more data in our applications, and therefore more markups. So our browsers need to handle much more computations than 10 years earlier. DOM operations are expensive, modern frameworks tend to reduce the number of DOM updates.
+
+### Why do we need utilities libraries
+
+As we saw earlier the way to handle immutability in EcmaScript is made simple thanks to syntactic sugar but it finds its limits in nested structures. With the arrival of libraries like redux, nested structures have become more popular.
```

Comment:
earlier => earlier,
but it finds its limits in nested structures => but is quite limited in nested structures
redux => `redux`